packages/sql-events/index.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+const MQEmitter = require('mqemitter')
+const fp = require('fastify-plugin')
+const { PassThrough } = require('stream')
+const MQEmitterRedis = require('mqemitter-redis')
+const { promisify } = require('util')
+
+async function fastifySqlEvents (app, opts) {
+  setupEmitter({ ...opts, mapper: app.platformatic, log: app.log })
+  const mq = app.platformatic.mq
+  app.addHook('onClose', () => promisify(mq.close.bind(mq)))
+}
+
+function setupEmitter ({ log, mq, mapper, connectionString }) {
+  if (connectionString) {
+    mq = MQEmitterRedis({ connectionString })
+  } else if (!mq) {
+    mq = MQEmitter()
+  }
+
+  for (const entityName of Object.keys(mapper.entities)) {
+    const entity = mapper.entities[entityName]
+    const { primaryKey } = entity
+    mapper.addEntityHooks(entityName, {
+      async save (original, data) {
+        const ctx = data.ctx
+        /* istanbul ignore next */
+        const _log = ctx?.reply?.request?.log || log
+        const res = await original(data)
+        const topic = await entity.getPublishTopic({ action: 'save', data: res, ctx })
+        if (topic) {
+          const payload = {
+            [primaryKey]: res[primaryKey]
+          }
+          _log.trace({ topic, payload }, 'publishing event')
+          await new Promise((resolve) => {
+            mq.emit({
+              topic,
+              payload
+            }, resolve)
+          })
+        }
+        return res
+      },
+
+      delete: multiField('delete'),
+      insert: multiField('save')
+    })
+
+    function multiField (action) {
+      return async function (original, data) {
+        const ctx = data.ctx
+        /* istanbul ignore next */
+        const _log = ctx?.reply?.request?.log || log
+        const fields = new Set(data.fields)
+        const res = await original({ ...data, fields: undefined })
```

Comment:
Getting all fields might cause significant unpredictable performance downgrade. Can we at least check if there is a subscription to this entity and do this only if it exists?

Reply:
Agreed, I'll remove and forward only the id to the subcriber.

Reply:
@mcollina Are you going to do it in this PR?

Reply:
This was done as part of https://github.com/platformatic/platformatic/pull/105/commits/f2e0e34b1af16293834cae58ce59d04ecd3ce36a